waku/protocol/v2/waku_protocol2.nim

```diff
@@ -68,10 +71,7 @@ method subscribe*(w: WakuSub,
   # XXX: Pubsub really
 
   # XXX: This is what is called, I think
-  if w.gossip_enabled:
-    await procCall GossipSub(w).subscribe(topic, handler)
-  else:
-    await procCall FloodSub(w).subscribe(topic, handler)
+  await procCall w.pubsub().subscribe(topic, handler)
```

Comment:
same here, `procCall` would simply never call your overloaded methods and go straigh to the base.

Reply:
```suggestion
  await procCall GossipSub(w).pubsub().subscribe(topic, handler)
```